presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -806,6 +806,13 @@ protected Scope visitJoin(Join node, Scope scope)
 
                 addCoercionForJoinCriteria(node, leftExpression, rightExpression);
                 expressions.add(new ComparisonExpression(EQUAL, leftExpression, rightExpression));
+
+                if (node.getType() == Join.Type.LEFT) {
+                    output = createScope(node, scope, left.getRelationType().joinWith(right.hide(rightExpression).getRelationType()));
```

Comment:
I typically use postgres to check the how given query behaves as I assume it has the best support of ANSI SQL within open source databases (am I right?). And it looks like it more complicated than I thought, please the below:

```
postgres=# select column1, t1.column1, t2.column1, * from t1 full outer join t2 using(column1);
 column1 | column1 | column1 | column1 | column2 | column2 
---------+---------+---------+---------+---------+---------
       1 |       1 |       1 |       1 |      -1 |     -11
       2 |       2 |         |       2 |      -2 |        
         |         |         |         |      -4 |        
       3 |         |       3 |       3 |         |     -33
         |         |         |         |         |     -44
(5 rows)

```

It looks like that `a.x` and `b.x` are hidden and `a` is equal to an expression `case a.x IS NULL THEN b.x ELSE a.x END`.